backend/src/test/java/com/woowacourse/gongcheck/core/application/JobServiceTest.java

```diff
@@ -65,272 +71,598 @@ class JobServiceTest {
-                () -> assertThat(taskRepository.findById(task.getId())).isEmpty(),
-                () -> assertThat(runningTaskRepository.findById(runningTask.getTaskId())).isEmpty()
-        );
-    }
+        @Nested
+        class findSlackUrl_메소드는 {
+            private final Long NON_EXIST_HOST_ID = 0L;
+            private final Long NON_EXIST_JOB_ID = 0L;
+
+            @Nested
+            class 존재하지_않는_Host의_id를_입력받은_경우 {
+
+                @Test
+                void 예외가_발생한다() {
+                    assertThatThrownBy(() -> jobService.findSlackUrl(NON_EXIST_HOST_ID, NON_EXIST_JOB_ID))
+                            .isInstanceOf(NotFoundException.class)
+                            .hasMessage("존재하지 않는 호스트입니다.");
+                }
+            }
+
+            @Nested
+            class Job이_존재하지_않는_경우 {
+
+                private Host host;
+                private final Long NON_EXIST_JOB_ID = 0L;
+
+                @BeforeEach
+                void setUp() {
+                    host = hostRepository.save(Host_생성("1234", 1234L));
+                }
+
+                @Test
+                void 예외가_발생한다() {
+                    assertThatThrownBy(() -> jobService.findSlackUrl(host.getId(), NON_EXIST_JOB_ID))
+                            .isInstanceOf(NotFoundException.class)
+                            .hasMessage("존재하지 않는 작업입니다.");
+                }
+            }
+        }
 
-    @Test
-    void Host가_존재하지_않는데_Slack_Url_조회_시_예외가_발생한다() {
-        assertThatThrownBy(() -> jobService.findSlackUrl(0L, 0L))
-                .isInstanceOf(NotFoundException.class)
-                .hasMessage("존재하지 않는 호스트입니다.");
-    }
+        @Nested
+        class Job이_존재하는데_다른_Host의_Job의_Slack_Url_조회하는_경우 {
+
+            private Host myHost;
+            private Job otherJob;
+
+            @BeforeEach
+            void setUp() {
+                myHost = hostRepository.save(Host_생성("1234", 1234L));
+                Host otherHost = hostRepository.save(Host_생성("1234", 2456L));
+                Space otherSpace = spaceRepository.save(Space_생성(otherHost, "잠실"));
+                otherJob = jobRepository.save(Job_생성(otherSpace, "톱오브스윙방"));
+            }
+
+            @Test
+            void 예외가_발생한다() {
+                assertThatThrownBy(() -> jobService.findSlackUrl(myHost.getId(), otherJob.getId()))
+                        .isInstanceOf(NotFoundException.class)
+                        .hasMessage("존재하지 않는 작업입니다.");
+            }
+        }
 
-    @Test
-    void Job이_존재하지_않는데_Slack_Url_조회_시_예외가_발생한다() {
-        Host host = hostRepository.save(Host_생성("1234", 1234L));
+        @Nested
+        class Job이_존재하는_경우 {
 
-        assertThatThrownBy(() -> jobService.findSlackUrl(host.getId(), 0L))
-                .isInstanceOf(NotFoundException.class)
-                .hasMessage("존재하지 않는 작업입니다.");
-    }
+            private Host host;
+            private Job job;
```

Comment:
```suggestion
            private static final String SLACK_URL = "http://slackurl.com";
            private Host host;
            private Job job;
```
조금 중요한 상수인듯해서 빼는게 좋을 것 같네요 :)